File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/DeviceEventFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment;

import android.os.Bundle;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.hp.workpath.api.Result;
import com.hp.workpath.api.device.events.DeviceEvent;
import com.hp.workpath.sample.deviceeventsample.Logger;
import com.hp.workpath.sample.deviceeventsample.R;
import com.hp.workpath.sample.deviceeventsample.task.DeviceEventTask;
import com.hp.workpath.sample.deviceeventsample.view.DeviceEventView;

import java.util.List;

import static com.hp.workpath.sample.deviceeventsample.DeviceEventActivity.INDEX;

public class DeviceEventFragment extends Fragment {

    LinearLayout layoutSummary;
    TextView mDeviceEventRawDataTextView;
    TextView mTitleTextView;
    ProgressBar mProgressBar;

    List<DeviceEvent> mDeviceEvents;
    DeviceEventView mDeviceEventView;

    int mIndex = 0;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_device_event, container, false);
        layoutSummary = view.findViewById(R.id.layoutSummary);
        mDeviceEventRawDataTextView = view.findViewById(R.id.deviceEventRawDataTextView);
        mDeviceEventRawDataTextView.setMovementMethod(new ScrollingMovementMethod());
        mTitleTextView = view.findViewById(R.id.titleTextView);
        mProgressBar = view.findViewById(R.id.progressBar);
        mDeviceEventView = new DeviceEventView(inflater, view);
        mIndex = getArguments().getInt(INDEX);
        getDeviceEvent();
        return view;
    }

    private void setRawData(DeviceEvent deviceEvent) {
        Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
        mDeviceEventRawDataTextView.setText(gson.toJson(deviceEvent));
    }

    public void getDeviceEvent() {
        mProgressBar.setVisibility(View.VISIBLE);
        new DeviceEventTask(getContext(), deviceEventInterface).taskExecute();
    }

    private void setDeviceEvent(DeviceEvent deviceEvent) {
        mTitleTextView.setText(deviceEvent.getTitle());
        setRawData(deviceEvent);
        mDeviceEventView.setDeviceEventData(deviceEvent);
    }

    ResponseInterface deviceEventInterface = new ResponseInterface() {
        @Override
        public void success(List<DeviceEvent> deviceEvents) {
            mProgressBar.setVisibility(View.GONE);
            mDeviceEvents = deviceEvents;
            if (mDeviceEvents != null && mDeviceEvents.size() > 0) {
                layoutSummary.setVisibility(View.VISIBLE);
                setDeviceEvent(mDeviceEvents.get(mIndex - 1));
            } else {
                mDeviceEventRawDataTextView.setText(getString(R.string.no_event));
                layoutSummary.setVisibility(View.GONE);
            }
        }

        @Override
        public void failure(String msg, Result result) {
            mProgressBar.setVisibility(View.GONE);
            Logger.showResult(getActivity(), msg, result);
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/DeviceEventFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment

import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import com.google.gson.GsonBuilder
import com.hp.workpath.api.Result
import com.hp.workpath.api.device.events.DeviceEvent
import com.hp.workpath.sample.deviceeventsample.DeviceEventActivity
import com.hp.workpath.sample.deviceeventsample.Logger
import com.hp.workpath.sample.deviceeventsample.R
import com.hp.workpath.sample.deviceeventsample.databinding.FragmentDeviceEventBinding
import com.hp.workpath.sample.deviceeventsample.task.DeviceEventTask
import com.hp.workpath.sample.deviceeventsample.view.DeviceEventView
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class DeviceEventFragment : Fragment() {

    private lateinit var mDeviceEvents: List<DeviceEvent>
    private lateinit var mDeviceEventView: DeviceEventView
    private var mBindingFragment: FragmentDeviceEventBinding? = null
    private val mBindingFragmentDeviceEvent get() = mBindingFragment!!
    var mIndex = 0

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mBindingFragment = FragmentDeviceEventBinding.inflate(inflater, container, false)
        val view = mBindingFragmentDeviceEvent.root
        mDeviceEventView = DeviceEventView(inflater, view)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mBindingFragmentDeviceEvent.deviceEventRawDataTextView.movementMethod =
            ScrollingMovementMethod()
        mIndex = arguments?.getInt(DeviceEventActivity.INDEX) ?: 0
        getDeviceEvent()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBindingFragment = null
    }

    private fun setRawData(deviceEvent: DeviceEvent) {
        val gson = GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
        mBindingFragmentDeviceEvent.deviceEventRawDataTextView.text = gson.toJson(deviceEvent)
    }

    private fun getDeviceEvent() {
        mBindingFragmentDeviceEvent.progressBar.visibility = View.VISIBLE
        lifecycleScope.launch(Dispatchers.Default) {
            DeviceEventTask(requireContext(), deviceEventInterface).execute()
        }
    }

    private fun setDeviceEvent(deviceEvent: DeviceEvent) {
        mBindingFragmentDeviceEvent.titleTextView.text = deviceEvent.title
        setRawData(deviceEvent)
        mDeviceEventView.setDeviceEventData(deviceEvent)
    }

    private var deviceEventInterface: ResponseInterface = object : ResponseInterface {
        override fun success(deviceEvents: List<DeviceEvent>?) {
            mBindingFragmentDeviceEvent.progressBar.visibility = View.GONE
            mDeviceEvents = deviceEvents as List<DeviceEvent>
            if (mDeviceEvents.isNotEmpty()) {
                mBindingFragmentDeviceEvent.layoutDeviceEvent.layoutSummary.visibility =
                    View.VISIBLE
                setDeviceEvent(mDeviceEvents[mIndex - 1])
            } else {
                mBindingFragmentDeviceEvent.deviceEventRawDataTextView.text =
                    getString(R.string.no_event)
                mBindingFragmentDeviceEvent.layoutDeviceEvent.layoutSummary.visibility = View.GONE
            }
        }

        override fun failure(msg: String?, result: Result?) {
            mBindingFragmentDeviceEvent.progressBar.visibility = View.GONE
            Logger.showResult(activity, msg, result)
        }
    }
}